bert_squeeze/data/modules/distillation_module.py

```diff
@@ -100,7 +100,7 @@ def _create_fake_label(example):
             return example
 
         soft_dataset = datasets.load_dataset(
-            self.soft_dataset_config.path, self.soft_dataset_config.split
+            self.soft_dataset_config.path,  # split=self.soft_dataset_config.split
         )
```

Comment:
Don't we want the `split` parameter set to `self.soft_dataset_config.split` here?

Reply:
well if you do this, when `TransformerDataModule.setup` is called, it breaks because it expects self.dataset to be a DataSetDict, via `featurize` as it picks train/validation/tests in it... this is why I commented out here. I think there's a type mismatch in that class, I am not what should be done. Maybe we should explicitely name things here (dataset vs dataset_dict) to clarify?

Reply:
sorry that comment was for the other class, but it is also valid for the Soft flavor. If you look at `get_soft_dataset` it looks for `["train"]`  which would break

Reply:
but maybe for that class we can make the assumption it's a dataset and always a train?

Reply:
> but maybe for that class we can make the assumption it's a dataset and always a train?

This makes sense to me

Reply:
I tried it won't work because this class has `featurize` which wants the `DatasetDict`.

I guess we have to keep dicts everywhere. and for the test, replace `train[:10%]` with a dict that returns 10% for each. I will try that